packages/compose/src/hooks/use-did-mount/index.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * WordPress dependencies
+ */
+import { useLayoutEffect, useRef } from '@wordpress/element';
+
+/**
+ * A drop-in replacement of the hook version of `componentDidMount`.
+ * Like `useEffect` but only called once when the component is mounted.
+ * This hook is only used for backward-compatibility reason. Consider using `useEffect` wherever possible.
+ *
+ * @param {Function} effect The effect callback passed to `useEffect`.
+ */
+function useDidMount( effect ) {
```

Comment:
I think `useMountEffect` or `useEffectOnMount` would be more descriptive names for this hook. I also don't think we should say it's "only used for backward-compatibility reason". There are plenty of valid use-cases to run something only once upon mount.

Reply:
Perhaps we should also create two versions of this hook... one for `useLayoutEffect` and one of `useEffect`. E.g. `useMountLayoutEffect` or `useLayoutEffectOnMount`. (I think the latter reads more clearly.)

Reply:
I disagree though, 99% of cases we can just use `useEffect` with an empty dependency array to achieve the same thing. If we're writing a new component, then I believe we can avoid using this hook if we can "think in hooks" in most cases. Besides, we can always adjust the wording if it does become more and more used!

Same thing goes for creating two versions of this hook. We can always create them if necessary. I find these hooks more like _utility_ hooks we can share between packages, but not like a standalone hooks library that's going to cover all cases. That's just my take though, happy to be proven wrong :)!

Reply:
Just want to share this tweet from React core team to back up my reasoning that making `useMount` more exposed might not be a good idea: https://twitter.com/sebmarkbage/status/1313842495256748038?s=21

Reply:
Thanks. Another React dev tweet worth sharing: https://twitter.com/dan_abramov/status/1284510166503890944